internal/db/schema/migrations/oss/postgres/64/01_ssh_targets.up.sql

```diff
@@ -0,0 +1,120 @@
+      check(session_max_seconds > 0),
+    -- limit on number of session connections allowed. -1 equals no limit
+    session_connection_limit int not null default 1
+      constraint session_connection_limit_must_be_greater_than_0_or_negative_1
+      check(session_connection_limit > 0 or session_connection_limit = -1),
+    create_time wt_timestamp,
+    update_time wt_timestamp,
+    version wt_version,
+    worker_filter wt_bexprfilter,
+    egress_worker_filter wt_bexprfilter,
+    ingress_worker_filter wt_bexprfilter,
+    constraint target_ssh_project_id_name_uq
+      unique(project_id, name) -- name must be unique within a project scope.
+  );
+  comment on table target_ssh is
+    'target_ssh is a table where each row is a resource that represents an ssh target. '
+    'It is a target subtype.';
+
+  drop trigger if exists insert_target_subtype on target_ssh;
+  create trigger
+    insert_target_subtype
+  before insert on target_ssh
+    for each row execute procedure insert_target_subtype();
+
+  drop trigger if exists delete_target_subtype on target_ssh;
+  create trigger
+    delete_target_subtype
+  after delete on target_ssh
+    for each row execute procedure delete_target_subtype();
+
+  -- define the immutable fields for target
+  drop trigger if exists immutable_columns on target_ssh;
+  create trigger immutable_columns before update on target_ssh
+    for each row execute procedure immutable_columns('public_id', 'project_id', 'create_time');
+
+  drop trigger if exists update_version_column on target_ssh;
+  create trigger update_version_column after update on target_ssh
+    for each row execute procedure update_version_column();
+
+  drop trigger if exists update_time_column on target_ssh;
+  create trigger update_time_column before update on target_ssh
+    for each row execute procedure update_time_column();
+
+  drop trigger if exists default_create_time_column on target_ssh;
+  create trigger default_create_time_column before insert on target_ssh
+    for each row execute procedure default_create_time();
+
+  drop trigger if exists update_ssh_target_filter_validate on target_ssh;
+  create trigger update_ssh_target_filter_validate before update on target_ssh
+    for each row execute procedure validate_filter_values_on_update();
+
+  drop trigger if exists insert_ssh_target_filter_validate on target_ssh;
+  create trigger insert_ssh_target_filter_validate before insert on target_ssh
+    for each row execute procedure validate_filter_values_on_insert();
+
+  insert into oplog_ticket (name, version)
+  values ('target_ssh', 1)
+  on conflict do nothing;
+
+  -- These views depend on target_all_subtypes, so to drop target_all_subtypes,
+  -- we need to drop these first. They will be recreated later on this migration
+  -- version (see 02_wh_sessions and 03_wh_ssh_cert_library).
+  drop view if exists whx_host_dimension_source;
+  drop view if exists whx_credential_dimension_source;
```

Comment:
These are not being created again within this migration file. Can we group things together so the drop and create happens in a single migration? That might mean combining this and the next two migrations, but that means we will only drop these once, and it is easier to confirm that it gets created again.

Reply:
That's fair, my only concern there is that the file would get too big but at least it offers that visibility.

Reply:
Addressed with [65815a6](https://github.com/hashicorp/boundary/pull/2960/commits/65815a6bfde660c3aa395cd25140e07d59f99693)